test/runnable/testinvariant.d

```diff
@@ -25,6 +25,76 @@ int testinvariant()
     return 0;
 }
 
+/***************************************************/
+// 519
+
+class C519
+{
+    invariant
+    {
+        ++x;
+    }
+    __gshared int x;
+}
+
+struct S519
+{
+    invariant()
+    {
+        printf("S519.invariant\n");
+        ++x;
+    }
+    __gshared int x;
+}
+
+struct S519c
+{
+    invariant()
+    {
+        printf("S519c.invariant\n");
+        ++x;
+    }
+    __gshared int x;
+    int y;
+}
+
+void test519()
+{
+    //printf("test1\n");
+    {
+        auto s = new C519();
+        assert(C519.x == 0);
+        delete s;
```

Comment:
`delete` is scheduled for deprecation.

Reply:
Even so, for this test to pass, you need to force call the dtor somehow, maybe by using `scope class`.  But that is allegedly deprecated also.  So ¯\_(ツ)_/¯

Reply:
I suspect `destroy(*foo)` should work instead of `delete foo` here, but this test seems to be failing regardless, so I couldn't verify.

Also, if you want to put the class on the stack, `scope auto s = new C519()` should work.  I'm currently contending that [`scope` in this use case is not being deprecated](https://github.com/dlang/dlang.org/pull/2027).

Reply:
Ah, I see that the original patch was incomplete.  `buildInv()` should be called before `buildDtor()` for classes too, not just structs.